src/awkward/_connect/julia/utils.py

```diff
@@ -0,0 +1,258 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+from __future__ import annotations
+
+import os
+import warnings
+from pathlib import Path
+
+from julia.api import JuliaError
+
+from awkward import __version__
+
+# FIXME: move this out?
+__awkward_array_jl_version__ = "1.0.0-DEV"
+
+juliainfo = None
+julia_initialized = False
+julia_kwargs_at_initialization = None
+julia_activated_env = None
+
+
+def _load_juliainfo():
+    """Execute julia.core.JuliaInfo.load(), and store as juliainfo."""
+    global juliainfo
+
+    if juliainfo is None:
+        from julia.core import JuliaInfo
+
+        try:
+            juliainfo = JuliaInfo.load(julia="julia")
+        except FileNotFoundError:
+            env_path = os.environ["PATH"]
+            raise FileNotFoundError(
+                f"Julia is not installed in your PATH. Please install Julia and add it to your PATH.\n\nCurrent PATH: {env_path}",
+            ) from None
+
+    return juliainfo
+
+
+def is_julia_version_greater_eq(juliainfo=None, version=(1, 9, 0)):
+    """Check if Julia version is greater than specified version."""
+    if juliainfo is None:
+        juliainfo = _load_juliainfo()
+    current_version = (
+        juliainfo.version_major,
+        juliainfo.version_minor,
+        juliainfo.version_patch,
+    )
+    return current_version >= version
+
+
+def _check_for_conflicting_libraries():  # pragma: no cover
+    """Check whether there are conflicting modules, and display warnings."""
+
+
+def _julia_version_assertion():
+    if not is_julia_version_greater_eq(version=(1, 9, 0)):
+        raise NotImplementedError(
+            "AwkwardArray requires Julia 1.9.0 or greater. "
+            "Please update your Julia installation."
+        )
+
+
+def _set_julia_project_env(julia_project, is_shared):
+    if is_shared:
+        os.environ["JULIA_PROJECT"] = "@" + str(julia_project)
+    else:
+        os.environ["JULIA_PROJECT"] = str(julia_project)
+
+
+def _process_julia_project(julia_project):
+    if julia_project is None:
+        is_shared = True
+        processed_julia_project = f"awkward-{__version__}"
+    elif julia_project[0] == "@":
+        is_shared = True
+        processed_julia_project = julia_project[1:]
+    else:
+        is_shared = False
+        processed_julia_project = Path(julia_project)
+    return processed_julia_project, is_shared
+
+
+def _import_error():
+    return """
+    Required dependencies are not installed or built.  Run the following command in your terminal:
+        python3 -m awkward install
```

Comment:
`pip install`?

Reply:
Is this error encountered if `pyjulia` is not properly installed? I found https://github.com/JuliaPy/pyjulia/blob/69b734f117e489904a07d20f46c8f96b56577779/src/julia/pytestplugin.py#L71-75 which suggests a message of
```
PyCall is not installed or built.  Run the following code in Python REPL:

    >>> import julia
    >>> julia.install()

See:
    https://pyjulia.readthedocs.io/en/latest/installation.html
```

Reply:
No, it's to install Julia dependencies for awkward, e.g. a specific Julia project -`AwkwardArray.jl` or check if it is installed.